object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -279,6 +306,276 @@ private static String getExactStringValue(Tree tree, ValueAnnotatedTypeFactory f
+            // the the AssignmentNode to the newDefs.
+            if ((rhs instanceof ObjectCreationNode)
+                || (rhs instanceof MethodInvocationNode)
+                || (rhs instanceof MethodAccessNode)
+                || (rhs instanceof LocalVariableNode
+                    && isVarInPrevDefs(newDefs, (LocalVariableNode) rhs))) {
+              newDefs.add((AssignmentNode) node);
+            }
+
+            // Ownership Transfer
+            if (rhs instanceof LocalVariableNode
+                && isVarInPrevDefs(newDefs, (LocalVariableNode) rhs)) {
+              newDefs.remove(getAssignmentNodeOfVar(newDefs, (LocalVariableNode) rhs));
+            }
+          }
+        }
+
+        // Remove the returned localVariableNode from newDefs.
+        if (node instanceof ReturnNode) {
+          Node result = ((ReturnNode) node).getResult();
+          if (result instanceof LocalVariableNode
+              && isVarInPrevDefs(newDefs, (LocalVariableNode) result)) {
+            newDefs.remove(getAssignmentNodeOfVar(newDefs, (LocalVariableNode) result));
+          }
+        }
+      }
+
+      for (BlockImpl succ : getSuccessors(state.first)) {
+        Set<AssignmentNode> toRemove = new HashSet<>();
+
+        // There is nothing needs to be checked
+        if (newDefs.size() == 0) {
+          propagate(Pair.of(succ, newDefs), visited, worklist);
+          continue;
+        }
+
+        CFStore succRegularStore = getAnalysis().getInput(succ).getRegularStore();
+
+        for (AssignmentNode assignmentNode : newDefs) {
+
+          // If the successor block is the exit block or if the set of the pair of block id and set
+          // of defs is repeated in a loop or if the variable is going out of scope
+          if (succ instanceof SpecialBlockImpl
+              || visited.contains(Pair.of(succ.getId(), newDefs))
+              || succRegularStore.getValue((LocalVariableNode) (assignmentNode.getTarget()))
+                  == null) {
+
+            if (nodes.size() == 0) { // If the cur block is special or conditional block
+              reportAlwaysCallErrors(assignmentNode, succRegularStore, null);
+
+            } else { // If the cur block is Exception/Regular block then it checks AlwaysCall
+              // condition in the store right after the last node
+              Node last = nodes.get(nodes.size() - 1);
+              CFStore storeAfter = getStoreAfter(last);
+              reportAlwaysCallErrors(
+                  assignmentNode,
+                  storeAfter,
+                  (last instanceof AssignmentNode) ? getAnnotatedType(last.getTree()) : null);
+            }
+          }
+
+          if (succRegularStore.getValue((LocalVariableNode) (assignmentNode.getTarget())) == null) {
+            // If the lhs of the assignmentNode isn't in the successor's input store, it means the
+            // variable is going out of scope
+            toRemove.add(assignmentNode);
```

Comment:
Can't this call go inside the previous `if` block?  Why do we need to check this condition again?